.github/workflows/test-canvas.yml

```diff
@@ -15,50 +15,59 @@ on:
       - '!src/canvas-extractor/**/*.md'
     branches:
       - main
+  workflow_dispatch:
+
 jobs:
   test-canvas-extractor:
     name: Run unit, style, and type checks
     runs-on: ubuntu-20.04
+    env:
+        PACKAGE_NAME: canvas-extractor
+        SRC_DIR: src/canvas-extractor
+        LOCK_FILE: src/canvas-extractor/poetry.lock
```

Comment:
Extracting these variables made it easy to copy and paste everything that follows into other workflows, without having to change anything. Unfortunately the variables cannot read from each other, so I had to duplicate myself. I think there is a way around this by creating a step that writes environment variables out to a file. I didn't worry about that at this time, as it took me a long time already to get the next steps right, and I was ready to move on.